desktop-app/app/menu.js

```diff
@@ -93,6 +94,21 @@ export default class MenuBuilder {
           });
         },
       },
+      {
+        label: 'Check for Updates...',
+        click() {
+          autoUpdater.checkForUpdatesAndNotify().then(r => {
```

Comment:
@jjavierdguezas `autoUpdater.checkForUpdatesAndNotify()` is already invoked from the `main` process during every load. 

Any idea what this will return if the update download is already completed from that call?

Reply:
I clicked several times the menu item (during and after the call to `checkForUpdatesAndNotify`  from app load) and always the same behavior: The same object (`r`) with the same info is returned and a notification is displayed (if the app is outdated). I really don't know if it downloads (and overwrite) the update every time or if it does something smart
From the [documentation](https://www.electron.build/auto-update):
>appUpdater.checkForUpdates() ⇒ Promise<UpdateCheckResult>
>Asks the server whether there is an update.
>
>appUpdater.checkForUpdatesAndNotify() ⇒ Promise< | UpdateCheckResult>
>Asks the server whether there is an update, download and notify if update available. 
>
>appUpdater.downloadUpdate(cancellationToken) ⇒ Promise<any>
>Start downloading update manually. You can use this method if autoDownload option is set to false.

Maybe, if we are going with a complete custom solution, we can call `checkForUpdates` instead and then call to `downloadUpdate` to download it manually then restart the app. but I think this is over-complicate a flow that is handled by `electron-updater`... what do you think @manojVivek  ?

Reply:
Yeah, let's not go that far. We should let `electron-updater` do it automatically.